### PR TITLE
updated doc string type for tally.nuclide

### LINKED ISSUE #2525: should tally.nuclides docstring request list of strings?
I was just making use of ```tally.nuclides``` and accidentally set it to a list of strings and delighted that it worked.

I notice the doc string asks for a list of ```openmc.Nuclide```

https://github.com/openmc-dev/openmc/blob/d8dbd9bfca5f9e8faf992422149ff0573fb49551/openmc/tallies.py#L59

Should this be change this to list of ```str```?

### PR DESCRIPTION
**Pull request by shimwell**
This PR closes #2525

### CODE REVIEW

**Review by pshriwise — APPROVED** [review]
Looks good! Thanks @shimwell!

**Review by paulromano — CHANGES_REQUESTED** [review]

**Review by paulromano — APPROVED** [review]

File: openmc/tallies.py [hunk 1]
```diff
@@ -56,7 +56,7 @@ class Tally(IDManagerMixin):
         Name of the tally
     filters : list of openmc.Filter
         List of specified filters for the tally
-    nuclides : list of openmc.Nuclide
+    nuclides : list of openmc.Nuclide or str
```
[the remark this answers is not in the document]
Comment (pshriwise) [reply openmc/tallies.py]: Whoops, sorry to lead you astray there @shimwell!

[the remark this answers is not in the document]
Comment (pshriwise) [reply openmc/tallies.py]: Though I'm a little confused. When was `openmc.Nuclide` deprecated?

[the remark this answers is not in the document]
Comment (shimwell) [reply openmc/tallies.py]: Not sure when but Paul did mention this deprecation during PR #2478 and I think ```element``` and ```macrosopic``` are also not used / no longer needed.

**Pull request merged**